wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +487,147 @@ def get_device_unique_id(device):
+        """
+        matches = []
+        for delegate_to in self._delegate:
+            if hasattr(delegate_to, attr):
+                matches.append(delegate_to)
+        if len(matches) == 1:
+            return getattr(matches[0], attr)
+        elif len(matches) > 1:
+            raise AttributeError("Ambiguous reference to attribute: {}".format(attr))
+        else:
+            raise AttributeError(attr)
+
+
+class LenovoSwitch(Entity):
+    """ Encapsulates all Lenovo Switches behavior """
+    API_PATH = "switches"
+
+    def __init__(self, switch_data, **kwargs):
+        super(LenovoSwitch, self).__init__(raw=switch_data, **kwargs)
+
+    @property
+    def product_name(self):
+        return self.raw.get("productName", None)
+
+    @property
+    def manufacturer(self):
+        return self.raw.get("manufacturer", None)
+
+    @property
+    def serial_number(self):
+        return self.raw.get("serialNumber", None)
+
+    @property
+    def part_number(self):
+        return self.raw.get("partNumber", None)
+
+    @property
+    def description(self):
+        return self.raw.get("description", None)
+
+    @property
+    def firmwares(self):
+        return self.raw.get('firmware', None)
+
+    @property
+    def power_status(self):
+        return self.raw.get('powerState', None).lower()
+
+    @property
+    def type(self):
+        return self.raw.get('type', None)
+
+    @property
+    def health_state(self):
+        return self.raw.get('overallHealthState', None)
+
+    @property
+    def ipv4_addresses(self):
+        return self.raw.get('ipv4Addresses', None)
+
+    @property
+    def ipv6_addresses(self):
+        return self.raw.get('ipv6Addresses', None)
+
+    @property
+    def ports(self):
+        return LenovoSwitch.get_switch_ports(self.raw)
+
+    @property
+    def ipv4_assignments(self):
+        return LenovoSwitch.get_switch_ipv4_assignments(self.raw)
+
+    @property
+    def ipv6_assignments(self):
+        return LenovoSwitch.get_switch_ipv6_assignments(self.raw)
+
+    @property
+    def _identifying_attrs(self):
+        return {"name": self.name, "uuid": self.uuid}
+
+    def refresh(self):
+        uri = "{}/{}".format(LenovoSwitch.API_PATH, self.name.lower())
+        switch_data = self.system._service_instance(uri)
+        self.raw = switch_data
+        return self.raw
+
+    @property
+    def name(self):
+        return self.raw.get("name", None)
+
+    @property
+    def uuid(self):
+        return self.raw.get("uuid", None)
+
+    def delete(self):
+        data = {"endpoints": [{
+                "ipAdresses": self.ipv4_addresses,
+                "type": self.type,
+                "uuid": self.uuid}],
+            "forceUnmanage": True}
+        self.system._service_post(path="unmanageRequest", request=data)
+
+    def cleanup(self):
+        self.delete()
+
+    @staticmethod
+    def get_switch_ipv4_assignments(switch_data):
```

Comment:
I'm not seeing the benefit of this combination of a `LenovoSwitch.ipv4assignments` property and a static method that takes the raw data available to a LenovoSwitch instance.

What's your intended use of these static methods, are there cases where you can't instantiate LenovoSystem and use `LenovoSystem().get_switch(<uuid>).get_ipv4_assignments` method that uses this logic but in a non-static method?